app/components/post_list/post/body/content/permalink_preview/permalink_preview.tsx

```diff
@@ -0,0 +1,197 @@
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            backgroundColor: theme.centerChannelBg,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.2),
+            borderWidth: 1,
+            borderRadius: 8,
+            marginTop: 8,
+            marginBottom: 8,
+            padding: 12,
+        },
+        header: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            marginBottom: 8,
+        },
+        authorInfo: {
+            flex: 1,
+            marginLeft: 12,
+            flexDirection: 'row',
+            alignItems: 'center',
+        },
+        authorName: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'SemiBold'),
+        },
+        timestamp: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+            marginLeft: 8,
+        },
+        messageContainer: {
+            marginBottom: 8,
+        },
+        messageText: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100),
+            lineHeight: 20,
+        },
+
+        channelContext: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+        },
+        channelName: {
+            color: theme.linkColor,
+            ...typography('Body', 75, 'SemiBold'),
+        },
+    };
+});
+
+const PermalinkPreview = ({embedData, showPermalinkPreviews, author, locale, teammateNameDisplay}: PermalinkPreviewProps) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    if (!showPermalinkPreviews) {
+        return null;
+    }
+
+    const {
+        post,
+        channel_display_name,
+        channel_type,
+    } = embedData;
+
+    const {formattedMessage} = useMemo(() => {
+        const message = post?.message;
+
+        if (!message || typeof message !== 'string') {
+            return {formattedMessage: ''};
+        }
+
+        const cleanMessage = message.trim();
+
+        const lines = cleanMessage.split('\n');
+        if (lines.length > 4) {
+            return {
+                formattedMessage: lines.slice(0, 4).join('\n') + '...',
+            };
+        }
+
+        return {
+            formattedMessage: cleanMessage,
+        };
+    }, [post?.message]);
+
+    const isEdited = useMemo(() => {
+        return post && post.edit_at && post.edit_at > 0;
+    }, [post]);
+
+    const channelContextText = useMemo(() => {
+        const channelPrefix = channel_type === 'D' ? '@' : '~';
+        return `${channelPrefix}${channel_display_name}`;
+    }, [channel_display_name, channel_type]);
+
+    const authorDisplayName = useMemo(() => {
+        if (author) {
+            return displayUsername(author, locale, teammateNameDisplay);
+        }
+
+        return post?.user_id ? `User ${post.user_id.slice(-4)}` : 'Unknown User';
```

Comment:
Again, this magic number of `4`.  We need to know what it's for.

Reply:
Added const.